### DISCUSSION TITLE
Azure Pipeline Migration Deploy

### DISCUSSION THREAD

**Discussion by danielweil** [prisma/prisma#13237] [Q&A]
Hello,

Thanks for the awesome library. I have a question.
How should we run the prisma migrations inside Azure Pipelines?

I was doing it with a bash task, but the pipeline does not have apparently access to Azure's Database. I tried using a service connection inside a Azure CLI task, but it didn't work as well.

My functions inside Azure has access to the database, but the Pipelines doesn't. I tried using it as a initial command inside functions app, but wasn't successfull.

Thanks for the help!

**Comment by ruheni** [comment]
Hi Daniel 👋🏽

Under the **Networking** settings of Azure SQL Server, check the "Allow Azure services and resources to access this server" checkbox to allow Azure services. This should be inclusive of Azure DevOps as it's a part of the services Azure provides.

You can then run `npx prisma migrate deploy`, to apply existing migrations to your database, as part of the build task of your application.

Let me know if you hit a snag. 🙂 

<img width="869" alt="Screenshot 2022-05-09 at 21 15 41" src="https://user-images.githubusercontent.com/33921841/167484545-967c6667-5431-4665-8b11-0a873a79279c.png">

**Comment by danielweil** [comment]
Hey. Thanks for the response!

I have a PostgreSQL instance at azure and cannot find the Networking option
in the sidebar. Do you know where can I find it?

Thanks!


Edit: Found it! Will ask for the TI team to enable it as I do not have permission for it.
Will get back to you!

**Reply by ruheni** [reply]
Hi Daniel 👋🏾 

Were you able to resolve the issue?

**Reply by P-A-R-U-S** [reply]
Yes, trying to do same with Postgress.

**Comment by johnnyoshika** [comment]
Related to this topic, I have an Azure postgres database that's secured with a virtual network with --subnet and --private-dns-zone. Had anyone figured out how to run prisma migration from an Azure DevOps pipeline when the database in configured like this?

**Reply by captain-derp** [reply]
Unless prisma can output an idempotent sql migration script like they do in .NET Entity Framework, it won't work.

**Reply by flo-sch** [reply]
You can define a [pipeline agent](https://learn.microsoft.com/en-us/azure/devops/pipelines/agents/agents?view=azure-devops&tabs=yaml%2Cbrowser) that runs in the same virtual network as your postgresql server.
That way you do not need to open public networking access to default MS hosted agents.

There are different options, I would recommend looking into "Managed DevOps Pool" for simplicity (while having capacity to run within a custom virtual network, as an option).

**Comment by captain-derp** [comment]
This may be considered a security issue because allowing public access or access from any Azure service is exposing your database to potential attacks. In order to allow migration for azure database that only allows access from the same VNet, Prisma should have a functionality to output a single migration scripts to apply migrations that has not been applied yet.

**Reply by janpio** [reply]
Feel free to open a feature request for this. First time I hear this suggestion, and if it is popular and sought-after we will definitely consider this.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
